docs/sources/reference/commandline/cli.md

```diff
@@ -760,6 +762,46 @@ you refer to it on the command line.
 > children) for security reasons, and to ensure repeatable builds on remote
 > Docker hosts. This is also the reason why `ADD ../file` will not work.
 
+Sometimes building an image will require access to certain environment variables
+that are specific to the build host like http-proxy; user credentials for
+pulling intermediate files etc. Most likely these variables can't be persisted
+in the built image for portatbility or security reasons i.e. it's not desirable
+to use the 'ENV' primitive of the Dockerfile.
+
+    $ docker build --build-env HTTP_PROXY=http://10.20.30.2:1234 .
+
+This will allow passing the built-time environment variables, that can be
+accessed like regular environment variables in the 'RUN' primitive of the
+Dockerfile, without persisting them in the intermediate or final images.
+
+> **Note:** environment variables defined using 'ENV' primitive of Dockerfile
+> will override the build-time environment variables.
+
+> **Note:** When desirable, variables passed from command-line can be persisted
+> in the final image by using ENV primitive and build-time environment variables
+> as shown below.
+
+    $ docker build --build-env CONT_IMG_VER=v1.0.1 - <<MARK
+    -> FROM ubuntu
+    -> ENV CONT_IMG_VER {$foo:-notSet}
+    -> MARK
+
+There are also cases where users would like to set the values of the variables
+defined in the Dockerfile, without having an explicit 'ENV' primitive.
+
+    $ docker build --build-env foo=bar .
+
+This will allow passing the value for a variable named `foo` and substitute all it's
+occurrences with value `bar` in Docker file primitives.
+
+> **Note:** the variable expansion is only supported for certain Dockerfile
+> primitives viz. ADD, COPY, ENV, EXPOSE, USER, VOLUME, WORKDIR
```

Comment:
It might be better to add a ref to https://docs.docker.com/reference/builder/#environment-replacement instead of trying to keep this list in-sync.